File: codigo/python/stat/graficos/diagrama-violin.py
## diagrama de Violìn

from pathlib import Path
import pandas as pd
import seaborn as sns
import matplotlib.pylab as plt

# leyendo data
DATA = Path().resolve() / 'data'
AIRLINE_STATS_CSV = DATA / 'airline_stats.csv'

airline_stats = pd.read_csv(AIRLINE_STATS_CSV)

# seaborn soporto el diagrama de violín con violinplot.

fig, ax = plt.subplots(figsize=(5, 5))
sns.violinplot(data=airline_stats, x='airline', y='pct_carrier_delay',
               ax=ax, inner='quartile', color='white')
ax.set_xlabel('')
ax.set_ylabel('Daily % of Delayed Flights')

plt.tight_layout()
plt.show()

File: ciencia-de-datos/code/statistics/r/stastics/graficos/diagrama-violin.R
## Diagrama de Violín
library(ggplot2)

# leyendo data
PSDS_PATH <- file.path(getwd())

airline_stats <- read.csv(file.path(PSDS_PATH, 'data',
 'airline_stats.csv'),
 stringsAsFactors = FALSE)

graph <- ggplot(data=airline_stats, aes(airline, pct_carrier_delay)) + 
  geom_violin(draw_quantiles = c(.25,.5,.75), linetype=2) +
  geom_violin(fill=NA, size=1.1) +
  coord_cartesian(ylim=c(0, 50)) +
  labs(x='', y='Daily % of Delayed Flights') +
  theme_bw()
graph

